libzeth/core/utils.tcc

```diff
@@ -13,6 +13,20 @@
 namespace libzeth
 {
 
+template<> constexpr size_t bit_utils<0>::bit_size() { return 0; }
+
+template<> constexpr size_t bit_utils<0>::num_true_bits() { return 0; }
+
+template<size_t X> constexpr size_t bit_utils<X>::bit_size()
+{
+    return 1 + bit_utils<(X >> 1)>::bit_size();
+}
+
+template<size_t X> constexpr size_t bit_utils<X>::num_true_bits()
+{
+    return (X & 1) + bit_utils<(X >> 1)>::bit_size();
```

Comment:
Hum.. Shouldn't it be `return (X & 1) + bit_utils<(X >> 1)>::num_true_bits();` here?

Reply:
Thanks.  Fixed this.

Reply:
Thanks. May be worth adding test cases (for these 2 functions) to https://github.com/clearmatics/zeth/blob/develop/libzeth/tests/core/utils_test.cpp as such things would have been easily caught by run `make check`

Reply:
Added test cases.